service/history/transferQueueProcessor.go

```diff
@@ -135,7 +138,7 @@ func (t *transferQueueProcessorImpl) Start() {
 		return
 	}
 	t.activeTaskProcessor.Start()
-	if t.isGlobalNamespaceEnabled {
+	if !t.singleCursor {
```

Comment:
there many places where you changed from checking isGlobalNamespaceEnabled to !singleCursor, it means if single cursor is not enabled (which is by default), you would run the logic that used to only run when global ns is enabled. Is that right?

Reply:
Sorry, the naming of this variable is confusing, it should be `singleProcessor`. The definition of it is on L95 and it's always true when global namespace is disabled. So those logic won't run when both global namespace and single cursor is disabled, which is the old behavior.